fw/cfg.h

```diff
@@ -191,6 +191,21 @@ typedef struct {
 	     (idx) < (e)->val_n;		\
 	     (v) = (e)->vals[++(idx)])
 
+#define TFW_CFG_CHECK_NO_ATTRS(spec, entry)				\
+	if ((entry)->attr_n) {						\
+		T_ERR_NL("%s: Arguments may not have the '=' sign\n",	\
+			 (spec)->name);					\
+		return -EINVAL;						\
+	}
+
+#define TFW_CFG_CHECK_VAL_N(op, req, spec, entry)				\
```

Comment:
I don't think we should implement this macros, we should use `tfw_cfg_check_val_n` in all places.

Reply:
`tfw_cfg_check_val_n` is more verbose and less declarative, I would prefer macros with `return` inside for that reason. But I can replace it, of cource.

Reply:
What is the reason that `tfw_cfg_check_val_n()` still exists and used in several places of Tempesta code? Or, the question rather should be: is there a clearly defined logic for whether the function or the new macro should be used? Right now it's inconsistent.

The new macro covers more cases. Yet, at the same time the macro is clearly used where the existing function would suffice. Yet more, like I said, the function is still used in several places and not replaced by the new macro.

What the new macro does well is that it makes checks short and concise in the code, and I like this refactoring a lot. It removes a lot of clutter. 

The other thing this refactoring does well is that it eliminates the disparity in return codes when checking the data. Not everything was covered by functions,  and where the functions were used `r` was returned, and in other places `-EINVAL` was returned explicitly. That created a visible disparity. It's less now that the return and the returned error are hidden inside the macro.

I am still ambivalent on the fact that it's a macro, but like I said it does provide for less clutter. Perhaps, other places with similar checks can be changed along the same lines? Not sure if that's a good idea but, perhaps, it's worth considering.

Reply:
Yes, it was unfinished refactoring, I refactored a lot of places, but not all. I think we need to define best way to work with it on the future meeting (in 2024). Pros: more declarative, cons: less composable, we can't combine several such macros (the reason of tfw_cfg_check_val_n existence).